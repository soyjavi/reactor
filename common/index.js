import { LAYOUT } from './layout/index';
import SHAPE from './shape';
import THEME from './theme';

export { LAYOUT, SHAPE, THEME };
