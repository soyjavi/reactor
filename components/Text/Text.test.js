import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import { Text } from '..';

const LOREM = 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.';

describe('<Text>', () => {
  it('renders', () => {
    const tree = renderer.create(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {headline}', () => {
    const tree = renderer.create(<Text headline>{LOREM}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {subtitle}', () => {
    const tree = renderer.create(<Text subtitle>{LOREM}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {caption}', () => {
    const tree = renderer.create(<Text caption>{LOREM}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {lighten}', () => {
    const tree = renderer.create(<Text lighten>{LOREM}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<Text color={THEME.COLOR.ERROR}>{LOREM}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {numberOfLines}', () => {
    const tree = renderer.create(<Text numberOfLines={2}>{LOREM}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<Text style={{ color: 'orange' }}>{LOREM}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('hook:styler', () => {
    const snapshot = renderer.create(<Text marginBottom="M" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
