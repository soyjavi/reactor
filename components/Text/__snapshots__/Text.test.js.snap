// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> renders 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontWeight": "400",
        "textAlignVertical": "center",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 19.2,
      },
      false,
      Object {},
    ]
  }
/>
`;

exports[`<Text> when {caption} 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontWeight": "400",
        "textAlignVertical": "center",
      },
      Object {
        "fontSize": 14,
      },
      false,
      Object {},
    ]
  }
>
  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
</Text>
`;

exports[`<Text> when {color} 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontWeight": "400",
        "textAlignVertical": "center",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 19.2,
      },
      false,
      Object {
        "color": "#ef5160",
      },
    ]
  }
>
  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
</Text>
`;

exports[`<Text> when {headline} 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontWeight": "400",
        "textAlignVertical": "center",
      },
      Object {
        "fontFamily": undefined,
        "fontSize": 24,
        "fontWeight": "700",
      },
      false,
      Object {},
    ]
  }
>
  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
</Text>
`;

exports[`<Text> when {lighten} 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontWeight": "400",
        "textAlignVertical": "center",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 19.2,
      },
      Object {
        "color": "#707070",
      },
      Object {},
    ]
  }
>
  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
</Text>
`;

exports[`<Text> when {numberOfLines} 1`] = `
<Text
  numberOfLines={2}
  style={
    Array [
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontWeight": "400",
        "textAlignVertical": "center",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 19.2,
      },
      false,
      Object {},
    ]
  }
>
  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
</Text>
`;

exports[`<Text> when {style} 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontWeight": "400",
        "textAlignVertical": "center",
      },
      Object {
        "fontSize": 16,
        "lineHeight": 19.2,
      },
      false,
      Object {
        "color": "orange",
      },
    ]
  }
>
  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
</Text>
`;

exports[`<Text> when {subtitle} 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
        "fontWeight": "400",
        "textAlignVertical": "center",
      },
      Object {
        "fontSize": 18,
        "fontWeight": "600",
      },
      false,
      Object {},
    ]
  }
>
  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
</Text>
`;
